presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/InlineProjections.java

```diff
@@ -156,6 +157,7 @@ private Expression inlineReferences(Expression expression, Assignments assignmen
                 .filter(entry -> entry.getValue() == 1) // reference appears just once across all expressions in parent project node
                 .filter(entry -> !tryArguments.contains(entry.getKey())) // they are not inputs to TRY. Otherwise, inlining might change semantics
                 .filter(entry -> !child.getAssignments().isIdentity(entry.getKey())) // skip identities, otherwise, this rule will keep firing forever
+                .filter(entry -> !(child.getAssignments().get(entry.getKey()) instanceof DereferenceExpression)) // skip dereference expression
```

Comment:
why is so?

Reply:
the goal of the PR is to put all dereference expressions in the first Project above TableScan. However, InlineProjections tries to push up all dereference expressions.